.enb/lib/page.js

```diff
@@ -0,0 +1,149 @@
+        browsers : browsers
+    }]);
+    node.mode('development', function() {
+        node.addTech([techs.files.copy, {
+            source : '.tmp.css',
+            target : '?.css'
+        }]);
+    });
+    node.mode('production', function() {
+        node.addTech([techs.process, {
+            source : '.tmp.css',
+            target : '?.css',
+            minify : true,
+            tech : 'cleancss'
+        }]);
+    });
+    node.addTarget('?.css');
+
+    // JavaScript for browsers
+    node.addTechs([
+        // Template engine for JavaScript
+        [techs.bem.depsByTechToBemdecl,  {
+            target : '.tmp.js-template.bemdecl.js',
+            sourceTech : 'js',
+            destTech : 'bemhtml'
+        }],
+        [techs.bem.deps, {
+            target : '.tmp.js-template.deps.js',
+            bemdeclFile : '.tmp.js-template.bemdecl.js'
+        }],
+        [techs.bem.files, {
+            depsFile : '.tmp.js-template.deps.js',
+            filesTarget : '.tmp.js-template.files',
+            dirsTarget : '.tmp.js-template.dirs'
+        }],
+        templateEngine === 'BH'? [techs.engines.bhClient, {
+            target : '.tmp.js-template.js',
+            filesTarget : '.tmp.js-template.files',
+            jsAttrName : 'data-bem',
+            jsAttrScheme : 'json',
+            mimic : 'BEMHTML'
+        }] : [techs.engines.bemhtml, {
+            target : '.tmp.js-template.js',
+            filesTarget : '.tmp.js-template.files',
+            devMode : false
+        }],
+
+        // JavaScript Files
+        [techs.bem.depsByTechToBemdecl, {
+            target : '.tmp.required-js.bemdecl.js',
+            sourceTech : 'js',
+            destTech : 'js'
+        }],
+        [techs.bem.mergeBemdecl, {
+            sources : ['.tmp.bemdecl.js', '.tmp.required-js.bemdecl.js'],
+            target : '.tmp.js.bemdecl.js'
+        }],
+        [techs.bem.deps, {
+            target : '.tmp.js.deps.js',
+            bemdeclFile : '.tmp.js.bemdecl.js'
+        }],
+        [techs.bem.files, {
+            depsFile : '.tmp.js.deps.js',
+            filesTarget : '.tmp.js.files',
+            dirsTarget : '.tmp.js.dirs'
+        }],
+
+        // JavaScript
+        [techs.browser.js, {
+            filesTarget : '.tmp.js.files',
+            target : '.tmp.source.js'
+        }],
+        [techs.browser.ym, {
+            source : '.tmp.source.js',
+            target : '.tmp.js'
+        }]
+    ]);
+    node.mode('development', function() {
+        node.addTech([techs.files.copy, {
+            source : '.tmp.js',
+            target : '?.js'
+        }]);
+    });
+    node.mode('production', function() {
+        node.addTech([techs.process, {
+            source : '.tmp.js',
+            target : '?.js',
+            minify : true
+        }]);
+    });
+    node.addTarget('?.js');
+
+    // Htmls by langs
+    node.addTechs([
+        templateEngine === 'BH'? [techs.engines.bhServer, {
```

Comment:
let's think about universal declaration for any template engine with the help of something like

```js
module.exports = {
    engine: 'bhServer',
    opts: {
        jsAttrName : 'data-bem',
        jsAttrScheme : 'json'
    }
}
```

and
```js
module.exports = {
    engine: 'bemhtml',
    opts: {
        devMode : false
    }
}
```

and then

```js
var templateEngineConf = require(templateEngine === 'BH'? `./bh.conf.js` : './bemhtml.conf.js');

[techs.engines[templateEngineConf.engine], extend({
    target : 'bemhtml.dev.js'
}, templateEngineConf.options)]
```

BTW, looks like there's no way to switch to dev mode in BEMHTML?